### ISSUE TITLE
Text result is split by spacing

### ISSUE THREAD

**Issue by Implocell** [jrmuizel/pdf-extract#79]
When running a pdf through it returns the text but in a weird spaced format. Is this intended or is it me that is doing something wrong? This just a random pdf from wikipedia pdf api

Code:
```rs
pub fn read_pdf() {
    let bytes = std::fs::read("2005_BBC_strike.pdf").unwrap();
    let out = pdf_extract::extract_text_from_mem(&bytes).unwrap();

    println!("{}", out);
}
```
Result:
```
2 0 0 5  B B C  s t r ik e

T he   2005  B B C   s tr ik e   w a s   a   s trike   of   m or e   tha n  1 1,000  B B C   w or ke rs ,  ove r  a   pr opos a l  to  c ut  4,000  jobs ,

a nd  to  pr iva tis e   pa rts   of   the   B B C   unde r  the   m a na ge m e nt   of   M a rk  T hom ps on . [ 1 ] [ 2 ]   M uc h  of   B B C 's   re gula r

pr ogr a m m ing  w a s   a ffe c te d,  w ith  m a ny   pr ogr a m s   be ing  re pl a c e d  w ith  pr e -re c or dings ,  a nd  s om e   be ing
```

**Comment by jrmuizel** [comment]
This isn't intended. Can you provide a link to the pdf?

**Comment by Implocell** [comment]
Here is the file in question, I've tried other files which does not come from wikipedia with the same result.
[2005_BBC_strike.pdf](https://github.com/jrmuizel/pdf-extract/files/13907260/2005_BBC_strike.pdf)

**Comment by Implocell** [comment]
Some files also return the following error: `unexpected smask type 933 0 R`
Do you want an example of one of these as well?
If there is anything I can do, please let me know

**Comment by jrmuizel** [comment]
Yes a link for the smask problem would be great too.

**Comment by Implocell** [comment]
This is created by me in Word (I think) and exported, just a simple line of text.
[TOP HEMMELIG.pdf](https://github.com/jrmuizel/pdf-extract/files/13907361/TOP.HEMMELIG.pdf)

**Comment by jrmuizel** [comment]
Do you see the smask problem in TOP.HEMMELIG.pdf? It works for me.

**Comment by jrmuizel** [comment]
It looks like there's a problem with the text transforms that might be causing this.

**Comment by jrmuizel** [comment]
Fixed by bf92c9b59c0bcc1d4a2ac1761dfd3fd913987987
